rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/IllegalOperationChecker.java

```diff
@@ -0,0 +1,142 @@
+                                   @BluetoothGattCharacteristicProperty int propertyWrite,
+                                   @BluetoothGattCharacteristicProperty int propertyNotify,
+                                   @BluetoothGattCharacteristicProperty int propertyIndicate,
+                                   @BluetoothGattCharacteristicProperty int propertySignedWrite,
+                                   IllegalOperationHandler resultHandler) {
+        this.propertyBroadcast = propertyBroadcast;
+        this.propertyRead = propertyRead;
+        this.propertyWriteNoResponse = propertyWriteNoResponse;
+        this.propertyWrite = propertyWrite;
+        this.propertyNotify = propertyNotify;
+        this.propertyIndicate = propertyIndicate;
+        this.propertySignedWrite = propertySignedWrite;
+        this.resultHandler = resultHandler;
+    }
+
+    /**
+     * This method checks whether the supplied characteristic possesses properties supporting the requested kind of operation, specified by
+     * the supplied bitmask.
+     *
+     * @param characteristic   a {@link BluetoothGattCharacteristic} the operation is done on
+     * @param neededProperties properties required for the operation to be successfully completed
+     * @return {@link Completable} deferring execution of the check till subscription
+     */
+    public Completable checkAnyPropertyMatches(final BluetoothGattCharacteristic characteristic,
+                                               final @BluetoothGattCharacteristicProperty int neededProperties) {
+        return Completable.fromCallable(new Callable<Object>() {
+            @Override
+            public Object call() throws Exception {
+                final int characteristicProperties = characteristic.getProperties();
+
+                if ((characteristicProperties & neededProperties) == 0) {
+                    int[] possibleProperties = getPossibleProperties();
+                    String message = String.format(
+                            Locale.getDefault(),
+                            "Characteristic %s supports properties:[%s] (%d) does not have any property matching [%s] (%d)",
+                            characteristic.getUuid(),
+                            propertiesIntToString(characteristicProperties, possibleProperties),
+                            characteristicProperties,
+                            propertiesIntToString(neededProperties, possibleProperties),
+                            neededProperties
+                    );
+                    resultHandler.handleMismatchData(message, characteristic.getUuid(), characteristicProperties, neededProperties);
+                }
+                return null;
+            }
+        });
+    }
+
+    @NonNull
+    private String propertiesIntToString(int property, int[] possibleProperties) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(" ");
```

Comment:
For prettyness. [ ] looks better than [] if there are no properties and [ READ WRITE ] looks better than [READ WRITE ]. I decided to avoid tracking whether any property has already been matched or not. I admit that I wrote specifically for the message which puts the string in square brackets, so for the purpose of extracting it to a separate parser, maybe the brackets should be added to this function?